### ISSUE TITLE
stop streaming button

### ISSUE THREAD

**Issue by rawwerks** [homanp/superagent#449]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/homanp/superagent/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/homanp/superagent/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Feature: A "Stop" Button for the chat streaming UX. (See https://chat.openai.com or https://poe.com for examples of products that have this.)

Benefit: Sometimes the LLM is too wordy and the stream rate is too slow. Allow the user to interrupt to stop.

### Acknowledgements

- [X] My issue title is concise and descriptive.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by homanp** [comment]
Thanks for adding
